app/src/marketplace/utils/smartContract.js

```diff
@@ -83,16 +83,17 @@ export const send = (method: Sendable, options?: {
 }): SmartContractTransaction => {
     const web3 = getWeb3()
     const emitter = new EventEmitter()
+    // NOTE: looks like there's double handling of errors happening here
+    // i.e. .catch + on('error')
```

Comment:
I wonder if `.on('error')` is "operational" at the point of that promise rejection (race condition). It's worth having a comment here though. Let it help the future us.